sdk/core/Azure.Core/src/RequestFailedException.cs

```diff
@@ -7,14 +7,13 @@ namespace Azure
 {
     public class RequestFailedException : Exception
     {
-        public const int NoStatusCode = 0;
         public int Status { get; }
 
-        public RequestFailedException(string message) : this(NoStatusCode, message)
+        public RequestFailedException(string message) : this(0, message)
```

Comment:
It seems odd to use 0, since 0 isn't a valid status code. Would it make sense to make Status a `int?` and default to null?

Reply:
It would force everyone catching the exception and wanting to check the status code to care about the `null` while exceptions without the status code would be in minority. In addition 0 not being a valid status code is a great reason to use it to represent no status code.

Reply:
Makes sense. 👍